moxin-frontend/src/landing/model_files_list.rs

```diff
@@ -228,3 +108,21 @@ impl ModelFilesListRef {
         inner.area.rect(cx).size.y
     }
 }
+
+fn model_featured_files(model: &ModelWithDownloadInfo) -> Vec<FileWithDownloadInfo> {
+    model
+        .files
+        .iter()
+        .filter(|f| f.file.featured)
+        .cloned()
```

Comment:
Nitpick: Is clone necessary or can we just return something like

```rust
fn model_featured_files(model: &'m ModelWithDownloadInfo) -> impl Iterator<Item = &'m FileWithDownloadInfo> {
    model
        .files
        .iter()
        .filter(|f| f.file.featured)
}
```

Reply:
Confirmed that `cloned` (not `clone`) is need here to compile it.

I think it is related with this part of the `cloned` docs:

> Creates an iterator which [clone](https://doc.rust-lang.org/std/clone/trait.Clone.html#tymethod.clone)s all of its elements.
> This is useful when you have an iterator over &T, but you need an iterator over T.

Reply:
> Confirmed that cloned (not clone)

Yeah, I was meaning if the action of cloning was necessary regardless of the name of the method implying the clone.

>  but you need an iterator over T.

Yeah, I mentioned this because I didn't notice a reason why this needs to be owned.

This is used just here right? https://github.com/moxin-org/moxin/blob/c1a8d2ef53eb859be1835c22a5f01a5cc1753691/moxin-frontend/src/landing/model_files_list.rs#L85

If you just need borrowed data you probably don't need ownership of the items. If it's being used just there, there shouldn't be lifetime issues. And if you are just accessing to the elements sequentially (`for i in 0..files_info.len()`) then anything implementing the `Iterator` trait should be able to be passes ad parameter.

(Of course what I mentioned will not compile if the other place is not changed).

But anyways, this is probably a less important comment that escaped my mind while reading the PR, and I didn't even tested this so please, proceed with merging this ignoring this less important comment.